### PR TITLE
add new react js example with vite

### PR DESCRIPTION
**Pull request by sanyamkamat**
## Description

Adding a working example of react-js app with Builder components.

_Screenshot_
image

### CODE REVIEW

**Review by samijaber — COMMENTED** [review]
I think the `package-lock.json` is missing here

File: examples/react-js/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+### To run the example Locally
+
+```bash
+git clone https://github.com/BuilderIO/builder.git
+cd examples/react-js
+npm install
+npm run dev
+```
+
+See src/main.jsx for the source
```
Comment (samijaber) [inline examples/react-js/README.md:20]: We need better instructions than that for new examples. I know you copied the one for the react one, but this is an old one and we want more details for our users.

Some examples:
- https://github.com/BuilderIO/builder/blob/main/examples/svelte/sveltekit/README.md
- https://github.com/BuilderIO/builder/blob/main/examples/next-js-simple/README.md


You can probably copy/paste a bunch of the stuff from the READMEs and replace command names.

File: examples/react-js/index.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Builder.io Plain JS example</title>
```
Comment (samijaber) [inline examples/react-js/index.html:4]:
```suggestion
    <title>Builder.io React JS example</title>
```

**Review by samijaber — COMMENTED** [review]
Thanks a lot for working on this @sanyamkamat . Only thing missing is a slightly improved README and then this is ready to go 🚀

File: examples/react-js/src/main.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+class CatchallPage extends React.Component {
+  state = { notFound: false };
+
+  render() {
+    return !this.state.notFound ? (
+      <BuilderComponent
+        apiKey="bb209db71e62412dbe0114bdae18fd15"
+        model="page"
+        contentLoaded={content => {
```
Comment (teleaziz) [inline examples/react-js/src/main.jsx:66]: Please replace the old react example, and fetch the content manually here, same as our quick start example on https://www.builder.io/c/docs/quickstart
```jsx
import { BuilderComponent, builder, useIsPreviewing } from '@builder.io/react';
import { useEffect, useState } from "react";

// Put your API key here
builder.init(YOUR_API_KEY)

// set whether you're using the Visual Editor,
// whether there are changes,
// and render the content if found
export default function CatchAllRoute() {
  const isPreviewingInBuilder = useIsPreviewing();
  const [notFound, setNotFound] = useState(false);
  const [content, setContent] = useState(null);

  // get the page content from Builder
  useEffect( () => {
    async function fetchContent() {
      const content = await builder.get('page', {
        url: window.location.pathname
      }).promise();

      setContent(content);
      setNotFound(!content);
    }
    fetchContent();
  }, []);
  // if no page is found, return a 404 page
  if (notFound && !isPreviewingInBuilder) {
    return <div>Not Found 404 Error</div>
  }

  // return the page when found
  return (
    <>
      {/* Render the Builder page */}
      <BuilderComponent model="page" content={content} />
    </>
  );
}

```

**Review by samijaber — APPROVED** [review]

**Pull request merged**